### DISCUSSION TITLE
axis centring

### DISCUSSION THREAD

**Discussion by thecuress** [martin2250/OpenCNCPilot#183] [General]
hello..... I use aspire to program the cut....in the image you can see the workpiece, which is very close to the x and y coordinates ... (you can see the distance)......In contrast, in the programme I find myself as in the image
![Immagine 1](https://github.com/martin2250/OpenCNCPilot/assets/58488797/ec0b2131-1951-4293-929d-e3f33d2a24b8)
![Immagine](https://github.com/martin2250/OpenCNCPilot/assets/58488797/e522277b-3d12-4f23-9f00-5a0258711ad0)
Is it possible to change the point x0y0z0 in opencncpilot ? thanks

**Comment by deHarro** [comment]
You can freely move around the viewport of OCP.
Just press and hold the middle mouse key and you can drag the workpiece around.
Pressing and holding the right mouse key lets you rotate the viewport.
If your view is as expected, use "Debug - Save Viewport Orientation" and OCP will remember these settings.

HTH
Harald

**Reply by thecuress** [reply]
I'm trying but it doesn't work,,,,,,if I press the centre button, a hand symbol appears, but if I position myself on the piece it doesn't move

**Comment by deHarro** [comment]
Mhmm...
If I try that myself, it just happens as described.
Know, that not the workpiece is moving relative to the grid, but the grid together with the workpiece moves around in the viewport.
So you should be able to move the workpiece together with the grid to the center of the viewport.
...
Ah!
Seemingly your workpiece is too big for the grid Martin implemented.
This is hardened by your screenshot, since one can see the edges of the grid. I've missed that fact at first.

@martin2250 should have a solution or at least a hint on what to do...

**Comment by thecuress** [comment]
yes, it goes off the grid

**Comment by deHarro** [comment]
Hi @martin2250 !

I just made a change to your code (locally at home :) to make the grid bigger for thecuress.
I think this is the correct location in file "MainWindow.xaml":

![grafik](https://github.com/martin2250/OpenCNCPilot/assets/26298406/ca779c16-e3d1-4c38-b778-310757a292ab)

I changed Width and Length from 500 to 800, respectively, and the grid in the viewport grew accordingly.

Is this the corrrect way to solve his problem?

**Comment by martin2250** [comment]
Hi everyone, Harald's solution is correct. Right now there is no way to change the size of the grid with a setting. When I work on OCP again, I can make this configurable in the settings window.

Cheers!

**Reply by deHarro** [reply]
Hi Martin,
I additionally created a fork with this Change and a Patch for better reading long lines in the file tab.
perhaps you can use that and make a version
Harald

**Reply by martin2250** [reply]
I'll check it out, thanks!

**Reply by deHarro** [reply]
I tried to make the file tab scalable with the mouse, but for that, the method of displaying the left bound boxes would have to be changed completely.
The net shows several threads with the same problem. They describe approaches, but way beyond my C# skills (... more or less nonexistent ☹).
